Hazel/src/Platform/Windows/WindowsWindow.cpp

```diff
@@ -172,5 +210,66 @@ namespace Hazel {
 	{
 		return m_Data.VSync;
 	}
+	
+	void WindowsWindow::SetWindowMode(const WindowMode& mode)
+	{
+		HZ_CORE_ASSERT(m_Window, "Failed to retrieve window.");
+		if (mode == m_Data.Mode) return;
+
+		GLFWmonitor* primaryMonitor = glfwGetPrimaryMonitor();
+		GLFWvidmode baseVideoMode = *(glfwGetVideoMode(primaryMonitor));
+
+		int windowPosX = 0, windowPosY = 0;
+		GLFWmonitor* monitor = nullptr;
+		int windowDecorated = GLFW_TRUE;
+		int windowResizeable = GLFW_TRUE;
+		int windowAutoIconify = GLFW_FALSE;
+
+		switch (mode)
+		{
+			default:
+				HZ_CORE_ERROR("WindowMode not supported.");
+			case WindowMode::Windowed:
+			{
+				m_Data.Width = m_Data.WindowedWidth;
+				m_Data.Height = m_Data.WindowedHeight;
+				windowPosX = m_Data.WindowedPos.x;
+				windowPosY = m_Data.WindowedPos.y;
+				break;
+			}
+			case WindowMode::Borderless:
+			{
+				m_Data.Width = baseVideoMode.width;
+				m_Data.Height = baseVideoMode.height;
+				monitor = primaryMonitor;
+				windowDecorated = GLFW_FALSE;
+				windowResizeable = GLFW_FALSE;
+				windowAutoIconify = GLFW_TRUE;
+				break;
+			}
+			case WindowMode::FullScreen:
+			{
+				// Achieved by creating a windowed window with size of monitor
+				if (m_Data.Mode == WindowMode::Borderless)
+					// TODO: temorary disable resize events
+					SetWindowMode(WindowMode::Windowed); // this way we obtain a window again
```

Comment:
To get to fullscreen from borderless, we first switch to windowed mode and let glfw handle the resizing. However, this will result in 2 resize callbacks:

1. Switching borderless -> windowed (unwanted)
2. Switching windowed -> fullscreen (wanted)

Solution: set callback to null, then switch to borderless, then reset the callback. When a previous callback was registered, this will be returned when calling the glfwSet*Callback function, so we can temporary store this.

Reply:
Implemented in https://github.com/TheCherno/Hazel/pull/108/commits/9ff06efb229f939f1f7b22d30dc55bb1449a2353